src/orion/storage/track.py

```diff
@@ -0,0 +1,464 @@
+            metrics=metrics
+        ))
+        return TrialAdapter(self.current_trial, objective=self.objective)
+
+    def register_lie(self, trial):
+        """Register a *fake* trial created by the strategist.
+
+        The main difference between fake trial and original ones is the addition of a fake objective
+        result, and status being set to completed. The id of the fake trial is different than the id
+        of the original trial, but the original id can be computed using the hashcode on parameters
+        of the fake trial. See mod:`orion.core.worker.strategy` for more information and the
+        Strategist object and generation of fake trials.
+
+        Parameters
+        ----------
+        trial: `Trial` object
+            Fake trial to register in the database
+
+        """
+        pass
+
+    def fetch_trials(self, query, *args, **kwargs):
+        """Fetch all the trials that match the query"""
+        def sort_key(item):
+            submit_time = item.submit_time
+            if submit_time is None:
+                return 0
+            return submit_time
+
+        query = to_json(query)
+
+        new_query = {}
+        for k, v in query.items():
+            if k == 'experiment':
+                new_query['group_id'] = v
+
+            elif k == 'heartbeat':
+                new_query['metadata.heartbeat'] = v
+
+            elif k == '_id':
+                new_query['uid'] = v
+
+            elif k == 'end_time':
+                new_query['metadata.end_time'] = v
+
+            elif k == 'status' and isinstance(v, str):
+                new_query['status'] = get_track_status(v)
+
+            else:
+                new_query[k] = v
+
+        trials = [
+            TrialAdapter(t, objective=self.objective) for t in self.backend.fetch_trials(new_query)
+        ]
+        trials.sort(key=sort_key)
+        return trials
+
+    _ignore_updates_for = {'results', 'params', '_id'}
+
+    def update_trial(self, trial, where=None, **kwargs):
+        """Update the fields of a given trials
+
+        Parameters
+        ----------
+        trial: Trial
+            Trial object to update
+
+        where: Optional[dict]
+            constraint trial must respect
+
+        kwargs: dict
+            a dictionary of fields to update
+
+        Returns
+        -------
+        returns true if the underlying storage was updated
+
+        """
+        try:
+            if isinstance(trial, TrialAdapter):
+                trial = trial.storage
+
+            for key, value in kwargs.items():
+                if key == 'status':
+                    self.backend.set_trial_status(trial, get_track_status(value))
+                elif key in self._ignore_updates_for:
+                    continue
+                else:
+                    pair = {key: to_json(value)}
+                    self.backend.log_trial_metadata(trial, **pair)
+
+            return True
+
+        except RuntimeError:
```

Comment:
Is this RuntimeError raises within Track? Would it make sense to define a custom error? I'm worried that we catch errors that we didn't intend to.